code/weapon/shockwave.cpp

```diff
@@ -297,32 +295,26 @@ void shockwave_move(object *shockwave_objp, float frametime)
 			}
 		}
 
-		// only apply damage to a ship once from a shockwave
-		for ( i = 0; i < sw->num_objs_hit; i++ ) {
-			if ( objp->signature == sw->obj_sig_hitlist[i] ){
-				break;
+		// only apply damage to a on object once from a shockwave if it's on the list
+		for (auto comparison : sw->obj_sig_hitlist) {
+			if ( (objp->signature == comparison.first) && (objp->type == comparison.second) ){
+				continue;
```

Comment:
This is only going to continue the comparison loop.  You want to set some flag here, break out of this loop, then continue the outer loop.  (That was the purpose of the `if ( i < sw->num_objs_hit )` in the original code.)